### ISSUE TITLE
Azure auth with service principal doc should not reference databricks user in example

### ISSUE THREAD

**Issue by atombravo** [databrickslabs/terraform-provider-databricks#1294]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Databricks Provider issue tracker reserved for bug reports and feature requests. For general Terraform questions, please see: https://www.terraform.io/community.html.
This template is for both adding enhancement as well as pointing out issues with the provider documentation.

### Affected Resource(s)
Link to the page with issues and a screenshot/description of what section has an issue.
Every doc page has anchor links for all the headings please post that here.

https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs#authenticating-with-azure-service-principal

### Expected Details
What are you expecting from the section that has an issue. If the section is missing anything that should be 
expected please point that out.

This section covers using an Azure service principal to authenticate, and provides the following example code which includes a "databricks_user" resource:

```
provider "azurerm" {
  client_id       = var.client_id
  client_secret   = var.client_secret
  tenant_id       = var.tenant_id
  subscription_id = var.subscription_id
}

resource "azurerm_databricks_workspace" "this" {
  location            = "centralus"
  name                = "my-workspace-name"
  resource_group_name = var.resource_group
  sku                 = "premium"
}

provider "databricks" {
  host                        = azurerm_databricks_workspace.this.workspace_url
  azure_workspace_resource_id = azurerm_databricks_workspace.this.id
  azure_client_id             = var.client_id
  azure_client_secret         = var.client_secret
  azure_tenant_id             = var.tenant_id
}

resource "databricks_user" "my-user" {
  user_name = "test-user@databricks.com"
}

```

### List of things to potentially add/remove:
This is a list of things to manipulate in the docs:

1. Remove the "databricks_user" resource in the example, since it's not relevant to authenticating with a service principal.
2. Add an example of a "databricks_service_principal" which is how you grant access to a service principal (by app ID instead of user@something.com syntax).

```
resource "databricks_service_principal" "example" {
  application_id = "00000000-0000-0000-0000-000000000000"
}

```
3. Add a link to the relevant documentation section for databicks_service_principal

### Important Factoids
Any links to external documentation that may prove your case, i.e Databricks public docs or Terraform public docs. 

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- #158

**Comment by nfx** [comment]
@atombravo please send a PR with the proposed fix. The easiest way would be by pressing "." (dot key) when you're in "docs" folder.

**Comment by alexott** [comment]
@atombravo it's a standard pattern to add initial user(s) (usually admins) when provisioning new workspace. Service principal will be automatically added into workspace on the first execution, so it's not strictly required to add it when you're doing this kind of authentication.

**Comment by atombravo** [comment]
@nfx just opened a PR, but I'm not sure how to tie it to this issue.

@alexott yeah, I realize now that the documentation in the authentication section is geared towards the initial setup of the TF environment and creation of the workspace, hence the example provided.  In my case, I'd created a workspace by running the TF under my own Azure credentials (with the az cli) and we attempting to switch to using a service principal for the existing object.  It wasn't obvious that the code is demoing how to create another (interactiive) user, -I think others might also interpret the example as creating a "user" that's the service principal under with the current credentials.  

I added a note explaining that a service principal creating the workspace would automatically get added, and to see the databricsks_service_principal for adding a principal to an existing workspace (such as I was trying to do).  I also added notation in the group and group_member documentation explicitly stating there that you can add a service principal to a group (and that you don't need, for example, to create a "user" that maps to a service principal to add it to a group).

**Comment by nfx** [comment]
merged #1298
